client/src/components/EditEvent.js

```diff
@@ -0,0 +1,228 @@
+import { useState, useEffect} from 'react';
+import { useHistory, useParams } from 'react-router-dom'
+
+function EditEvent () {
+
+    const history = useHistory()
+    const { id } = useParams()
+
+    const [title, setTitle] = useState('')
+    const [isvirtual, setVirtual] = useState(false)
+    const [location, setLocation] = useState('')
+    const [description, setDescription] = useState('')
+    const [time_start, set_time_start] = useState('')
+    const [time_end, set_time_end] = useState('');
+
+    function dateInputParser (rawDate) {
+        const obj = new Date(rawDate)
+        const year = obj.getFullYear()
+        let month = obj.getMonth() + 1
+        month = String(month).length === 1 ? `0${month}` : month
+        let day = obj.getDate()
+        day = String(day).length === 1 ? `0${day}` : day 
+        let hours = obj.getHours()
+        hours = String(hours).length === 1 ? `0${hours}` : hours 
+        let minutes = obj.getMinutes()
+        minutes = String(minutes).length === 1 ? `0${minutes}` : minutes
+        return `${year}-${month}-${day}T${hours}:${minutes}`
+    }
+
+    useEffect(() => {
+        fetch(`${process.env.REACT_APP_API_URL}/events/${id}`)
+        .then(res => {
+            if (res.ok) {
+                res.json().then(json => {
+                    setTitle(json.title)
+                    setVirtual(json.isvirtual)
+                    setLocation(json.location)
+                    setDescription(json.description)
+                    set_time_start(json.time_start)
+                    set_time_end(json.time_end)
+                })
+            } else {
+                console.log('there was an error')
```

Comment:
Maybe throw an error instead?

Reply:
Planned to do another PR just for errors down the road, for now they're just a bandaid, but a necessary bandaid